doc/build-helpers/testers.chapter.md

```diff
@@ -421,6 +421,11 @@ Check that two paths have the same contents.
 
 : A message that is printed last if the file system object contents at the two paths don't match exactly.
 
+`checkMetadata` (boolean)
+
+: Whether to fail on metadata differences, such as permissions or ownership.
+  Defaults to `true`.
```

Comment:
Maybe this should default to `stdenv.hostPlatform.isLinux`? Based on your constant usage of it and the justified reasoning in the commit message.